transformations/negate_strengthen/transformation.py

```diff
@@ -0,0 +1,462 @@
+            if None not in method:
+                t_dict[jx] = {
+                    'edit_text': edit_text,
+                    'edit_id': edit_id,
+                    'edit_method': method,
+                    'alt_word': None,
+                    'alt_text': None
+                }
+
+        return t_dict
+
+
+    def generate(self, sentence: Union[dict, str], target: str):
+
+        if isinstance(sentence, dict):
+            # You can indicate specific word location with var "get_roots" as a list of index
+            # especially if you already know the root word you wish to negate/strengthen upon. 
+            # E.g. AltLex dataset has "signal_id" info
+            get_roots = [sentence['signal_id']]
+            sentence = sentence['sentence']
+        else:
+            # If unspecified, we revert to Root Verbs based on spacy dependency parser.
+            get_roots = None
+
+        if self.verbose:
+            print('>>>>>>> sentence: ',sentence)
+            print('>>>>>>> target: ',target)
+
+        sentid2tid = {0: 1} # fixed since only one-sentence per doc
+        perturbed_sentences = ["NA"]
+        perturbed_target = "NA"
+        
+        if target in ['Direct Causal', 'Direct Relation']:
+            # Negation: Direct Causal -> No Relationship
+            t_dict = self.text_to_negated_edits(sentence, sentid2tid, get_roots=get_roots)
+            if self.verbose:
+                print(">>>>>> t_dict: ", t_dict)
+
+            if len(t_dict)>0:
+                # Found available negation
+                perturbed_sentences = []
+                for k, v in t_dict.items():
+                    # Select edit or alt edit
+                    perturbed_sentence = self.format_text(self.select_edit(v))
+                    # Store output
+                    perturbed_sentences.append(perturbed_sentence)
+                perturbed_target = 'No Relationship'
+
+        elif target == 'Conditional Causal':
+            # Strengthen: Conditional Causal -> Direct Causal
+            t_dict = self.text_to_stronger_edits(sentence, sentid2tid)
+            if self.verbose:
+                print(">>>>>> t_dict: ", t_dict)
+
+            if len(t_dict)>0:
+                # Found available strengthening
+                perturbed_sentences = []
+                for k, v in t_dict.items():
+                    # Format sentence
+                    perturbed_sentence = self.format_text(v['edit_text'])
+                    # Store output
+                    perturbed_sentences.append(perturbed_sentence)
+                perturbed_target = 'Direct Causal'
+        else:
+            # Any future conversion implementations
+            pass
+        
+        # Current NLAugmenter set up makes it hard to check more than 1 edit while checking in order
+        # for now, we only keep the first possible edit per example
+        perturbed_items  = [(perturbed_sentences[0], perturbed_target)]
```

Comment:
NL-Augmenter now supports returning multiple transformed texts. The returned variable is a list of dictionary.